File: feedback-triage-frontend/src/app/core/models/FeedbackResponse.ts
import { Channel } from "../enums/Channel";
import { FeedbackStatus } from "../enums/FeedbackStatus";
import { Sentiment } from "../enums/Sentiment";
import { UrgencyLevel } from "../enums/UrgencyLevel";

export interface FeedbackResponse {
  content:          Content[];
  empty:            boolean;
  first:            boolean;
  last:             boolean;
  number:           number;
  numberOfElements: number;
  pageable:         Pageable;
  size:             number;
  sort:             Sort;
  totalElements:    number;
  totalPages:       number;
}

export interface Content {
  feedbackId: string;
  message: string;
  emailContact: string;
  channel: Channel;
  urgency: UrgencyLevel;
  sentiment: Sentiment;
  category: string;
  needsReview: boolean;
  status: FeedbackStatus;
  classifiedAt: string;
}

export interface Pageable {
  offset:     number;
  pageNumber: number;
  pageSize:   number;
  paged:      boolean;
  sort:       Sort;
  unpaged:    boolean;
}

export interface Sort {
  empty:    boolean;
  sorted:   boolean;
  unsorted: boolean;
}

File: dashboard-service/src/main/java/com/franrl/dashboard_service/dto/FeedbackResponse.java
package com.franrl.dashboard_service.dto;

import com.franrl.dashboard_service.entity.FeedbackEntity;
import com.franrl.dashboard_service.entity.FeedbackStatus;
import com.franrl.enums.Channel;
import com.franrl.enums.Sentiment;
import com.franrl.enums.UrgencyLevel;

import java.time.Instant;
import java.util.UUID;

public record FeedbackResponse(
        UUID feedbackId,
        String message,
        String emailContact,
        Channel channel,
        UrgencyLevel urgency,
        Sentiment sentiment,
        String category,
        Boolean needsReview,
        FeedbackStatus status,
        Instant classifiedAt
) {

    public static FeedbackResponse fromEntity(FeedbackEntity entity) {
        return new FeedbackResponse(
                entity.getFeedbackId(),
                entity.getMessage(),
                entity.getEmailContact(),
                entity.getChannel(),
                entity.getUrgency(),
                entity.getSentiment(),
                entity.getCategory(),
                entity.getNeedsReview(),
                entity.getStatus(),
                entity.getClassifiedAt()
        );
    }
}
